packages/backend/src/graphql/mutations/tiles/update-table.ts

```diff
@@ -40,26 +44,48 @@ const updateTable = async (
   }
 
   if (addedColumns?.length) {
-    await table
-      .$relatedQuery('columns')
-      .insert(addedColumns.map((name) => ({ name })))
+    if (addedColumns.some((name) => !name.trim().length)) {
+      throw new Error('Invalid column name')
+    }
+    await TableColumnMetadata.transaction(async (trx) => {
+      const results = await table
+        .$relatedQuery('columns', trx)
+        .max('position as position') // aliasing for more convenient typing
+      const maxPosition = results[0].position || 0
+      await table.$relatedQuery('columns', trx).insert(
+        addedColumns.map((name, i) => ({
+          name,
+          position: maxPosition + i + 1,
+        })),
+      )
+    })
   }
 
   if (modifiedColumns?.length) {
-    await Promise.all(
-      modifiedColumns.map(async (column) => {
-        await table
-          .$relatedQuery('columns')
-          .patchAndFetchById(column.id, {
-            name: column.name,
-          })
-          .throwIfNotFound()
-      }),
-    )
+    await TableColumnMetadata.transaction(async (trx) => {
+      await Promise.all(
```

Comment:
nit: should monitor this a little bit, i think this can be a single query instead of a promise.all. Main concern is tables with a ton of columns (can look at some of the vault workspace pipes) blasting like 40+ queries in a single update.

Reply:
addressed